src/go/guestagent/pkg/tracker/apitracker.go

```diff
@@ -92,9 +88,21 @@ func (a *APITracker) Add(containerID string, portMap nat.PortMap) error {
 				continue
 			}
 
-			log.Debugf("calling %s API for the following port binding: %+v", exposeAPI, portBinding)
+			if a.enableListeners {
+				hostPort, err := strconv.Atoi(portBinding.HostPort)
+				if err != nil {
+					log.Errorf("error converting hostPort: %s", err)
+					continue
+				}
+				if err := a.AddListener(a.context, net.IP(portBinding.HostIP), hostPort); err != nil {
+					log.Errorf("creating listener for %s and %s failed: %s", portBinding.HostIP, portBinding.HostPort, err)
+					continue
+				}
+			}
 
-			err = a.expose(
+			log.Debugf("calling /services/forwarder/expose API for the following port binding: %+v", portBinding)
```

Comment:
Seems odd to hard code the path in the debug message here.
```suggestion
			log.Debugf("exposing the following port binding: %+v", portBinding)
```